hash_practice/exercises.py

```diff
@@ -5,17 +5,56 @@ def grouped_anagrams(strings):
         Time Complexity: ?
         Space Complexity: ?
     """
-    pass
+    if len(strings) == 0:
+        return []
+    if len(strings) == 1:
+        return [[strings[0]]]
+    list1 = [strings[0]]
+    listOfLists = [list1]
+    for word in strings[1:]:
+        flag = False
+        for lists in listOfLists:
+            # check in the existing inner lists
+            if anagram_helper(word) == anagram_helper(lists[0]):

```

Comment:
This solution redoes a great deal of work as it recalculates the frequency map through `anagram_helper` again and again for each word in the list.

Additionally going back over every list makes this algorithm balloon to O(n^2) in time complexity since in the worst case, every word is in its own list and there are none that share anagrams.

I think you can achieve a much better solution that is O(n) through using a dictionary, because with this solution you don't gain advantage of the O(1) lookup time of dictionaries.